sdk/instrumentation/opentelemetry-instrumentation-azure-sdk/src/configuration.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export const SDK_VERSION: string = "1.0.0-beta.2";
+
+/**
+ * @internal
+ *
+ * Keys of known environment variables we look up.
+ */
+export type KnownEnvironmentKey = "AZURE_HTTP_TRACING_DISABLED" | "AZURE_TRACING_DISABLED";
+
+export const environment = new Map<KnownEnvironmentKey, string | undefined>();
+
+if (typeof process !== "undefined" && process.env) {
+  environment.set("AZURE_HTTP_TRACING_DISABLED", process.env.AZURE_HTTP_TRACING_DISABLED);
+  environment.set("AZURE_TRACING_DISABLED", process.env.AZURE_TRACING_DISABLED);
+}
+
+/**
+ * Converts an environment variable to Boolean.
+ * the strings "false" and "0" are treated as falsy values.
+ *
+ * @internal
+ */
+export function envVarToBoolean(key: KnownEnvironmentKey): boolean {
+  const value = (environment.get(key) ?? "").toLowerCase();
```

Comment:
by caching the environment, are we making it harder to test externally? You can override `process.env` dynamically within the node process

Reply:
It's definitely a valid point, Liudmila also had a good point about the perf impact of fetching env vars on the code's hot path which is why I cached them 

https://github.com/Azure/azure-sdk-for-js/pull/20776#discussion_r824291453
https://github.com/Azure/azure-sdk-for-js/pull/20776#discussion_r824295791

I am admittedly just learning about this perf impact but it does sound like a valid concern - what do you think?

Reply:
I'm not opposed to caching, though I would imagine this would be one level up, so we'd cache the value of these two specific values after they've been accessed?

Reply:
Makes sense, I can lazy-read them and cache on first access instead 👍